### PR TITLE
feat(widget): widgets are communicating via service (DEV-1643)

### PR DESCRIPTION
**Pull request by Vijeinath**
resolves DEV-1643

There are two buttons on the page of MLS (localhost:4200) which demonstrate the two cases of communication:
 1. `Change Background` as synchronous event
 2. `Search` as asynchronous event

### CODE REVIEW

**Comment by Vijeinath** [comment]
I had to change a lot of files just to display the two widgets. Maybe in the future we should provide a structure which makes it easy to integrate widgets later on.

File: src/app/services/component-communication.service.ts [hunk 1]
```diff
@@ -13,14 +13,18 @@ export class ComponentCommunicationService {
...
-    on(event: Events, action: (value: any) => void): Subscription {
+    on(event: Events, action: (value: any) => void, loading?: () => void): Subscription {
         return this._subject$
             .pipe(
                 // filter down based on event name to any events that are emitted out of the subject from the emit method below.
                 filter((e: CpeEvent, i: any) => e.name === event),
-                map((e: CpeEvent) => e.value)
-            )
-            .subscribe(action); // subscribe to the subject to get the data.
+            ).subscribe((e: CpeEvent) => {
```
Comment (mdelez) [inline src/app/services/component-communication.service.ts:21]: I like this! Can you also update the comment to match?

File: src/app/services/component-communication.service.ts [hunk 2]
```diff
@@ -31,11 +35,16 @@ export class ComponentCommunicationService {
...
 
 // possible events that can be emitted.
 export enum Events {
-    searchExecuted
+    searchExecuted,
+    changeBackground
 }
 
+export enum Status {
+    starting,
```
Comment (mdelez) [inline src/app/services/component-communication.service.ts:48]: maybe we should stick to past tense, so in this case "started"

**Review by mdelez — CHANGES_REQUESTED** [review]
Nice changes :) Just a few small remarks

File: src/app/services/component-communication.service.ts (hunk 1 above)
Comment (Vijeinath) [reply src/app/services/component-communication.service.ts]: Done in https://github.com/dasch-swiss/dsp-cpe/pull/11/commits/ce69a805ba4a8febc3b83d1ebc2d204ee1071542

File: src/app/services/component-communication.service.ts (hunk 2 above)
Comment (Vijeinath) [reply src/app/services/component-communication.service.ts]: That sounds better -> https://github.com/dasch-swiss/dsp-cpe/pull/11/commits/ce69a805ba4a8febc3b83d1ebc2d204ee1071542

**Comment by Vijeinath** [comment]
@mdelez I also added an error status because the component also should react to that event. You can see a third button and test it

File: src/app/cpe/widgets/search-widget/search-widget.component.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<div>Search widget</div>
+<br>
+<cpe-button
+    [id]="'button-01'"
```
Comment (mdelez) [inline src/app/cpe/widgets/search-widget/search-widget.component.html:4]: from what I can tell, `id` is not an input in the `cpe-button` component so the id shouldn't be wrapped in square brackets

**Review by mdelez — CHANGES_REQUESTED** [review]

File: src/app/cpe/widgets/search-widget/search-widget.component.html (hunk 3 above)
Comment (Vijeinath) [reply src/app/cpe/widgets/search-widget/search-widget.component.html]: Done in https://github.com/dasch-swiss/dsp-cpe/pull/11/commits/ec92e796e8dcb7b4f872614f50db5574523e988b

**Review by mdelez — APPROVED** [review]
Thanks!

**Pull request merged**